tests/queries/0_stateless/02403_enable_extended_results_for_datetime_functions.reference

```diff
@@ -54,3 +70,19 @@ toLastDayOfMonth;toDate32;false	1970-01-01
 type;toLastDayOfMonth;toDate32;false	Date
 toLastDayOfMonth;toDateTime64;false	1970-01-01
 type;toLastDayOfMonth;toDateTime64;false	Date
+toStartOfDay;toDateTime64;false	1970-01-01 00:00:00
+type;toStartOfDay;toDateTime64;false	DateTime
+toStartOfHour;toDateTime64;false	1970-01-01 00:00:00
+type;toStartOfHour;toDateTime64;false	DateTime
+toStartOfMinute;toDateTime64;false	1970-01-01 00:00:00
+type;toStartOfMinute;toDateTime64;false	DateTime
+toStartOfFiveMinutes;toDateTime64;false	2056-03-09 16:48:16
```

Comment:
Not sure that I understand why '16:48:16' is expected for toStartOfFiveMinutes

Reply:
This is because "narrowing" is not implemented for toStartOfFiveMinutes, toStartOfTenMinutes, toStartOfFifteenMinutes and timeSlot function when the option enable_extended_results_for_datetime_functions is OFF. When the option is ON it works fine. @rschu1ze What do you think about it? Is it OK for current PR? Or should add narrowing for these functions in current PR for the case when the option is OFF.

Reply:
From a correctness POV, the current behavior is not great. But I think that the compatibility POV is more important here, so we better don't break existing behavior. At some point in future, `enable_extended_results_for_datetime_functions` will be `ON` by default, and we can stop worrying about this issue.

Reply:
OK. We will keep it as it is for now.